knowledge-base/how-tos/remote-debugging.md

```diff
@@ -0,0 +1,46 @@
+# How to remote debug an application on SAP Cloud Platform
+
+## TLDR;
+
+Change your app to start to start in debug mode, ssh tunnel your local port 9229 to the remote port of 9229 of your application and start a remote debugging session from your IDE.
+
+## Steps
+
+### Deploy your application with debug mode
+
+* Replace the start script to run in debug mode. E. g. instead of `node start.js` use `node --inspect start.js` for node and instead of `ts-node start.ts` use `node -r ts-node/register --inspect start.ts`.
+  **Note:** Don't use `--inspect-brk`, this will make the start timeout on SAP Cloud Platform.
+
+Deploy your application as usual, run:
+```shell
+cf deploy
+```
+
+### Open an ssh tunnel to your application
+Open an ssh tunnel to your backend application to connect your local debugger with the node inspector running on port 9229. Replace *<your-app-name>* with your application name and run:
+```shell
+$ cf ssh <your-app-name> -L 9229:127.0.0.1:9229 -T -N
+```
+### Attach a local debugger
+Now you can attach your local debugger. For this you will have to launch a debugger that attaches to the remote session. In VSCode this is a launch configuration you can use, when you replace *<path-to-your-application>* with the relative path to your application directory:
+```json
```

Comment:
Perhaps you can comment or add this screenshot:
![image](https://user-images.githubusercontent.com/56544999/98130481-95beec80-1eba-11eb-9008-61b6e721887e.png)
for the webstorm users.

Reply:
Done.